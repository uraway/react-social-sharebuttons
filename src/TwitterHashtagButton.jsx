// @flow
import React, { Component } from 'react';

declare var twttr: any;

type Props = {
  hashtag: string,
  text?: string,
};

type State = {
  initialized: boolean,
};

export default class TwitterHashtagButton extends Component<Props, State> {
  static defaultProps = {
    text: '',
  };

  node = null;

  constructor(props: Props) {
    super(props);
    this.state = { initialized: false };
  }

  componentDidMount() {
    const { initialized } = this.state;
    if (initialized) {
      return;
    }

    if (typeof twttr === 'undefined') {
      const twitterbutton = this.node;
      const twitterscript = document.createElement('script');
      twitterscript.src = '//platform.twitter.com/widgets.js';
      twitterscript.async = true;
      twitterscript.id = 'twitter-wjs';
      if (twitterbutton && twitterbutton.parentNode) twitterbutton.parentNode.appendChild(twitterscript);
    } else {
      twttr.widgets.load();
    }

    this.initialized();
  }

  initialized() {
    this.setState({ initialized: true });
  }

  render() {
    const { hashtag, text } = this.props;
    return (
      <a
        ref={(node) => (this.node = node)}
        href={`https://twitter.com/intent/tweet?button_hashtag=${hashtag}&text=${text || ''}`}
        className="twitter-hashtag-button"
      >
        Tweet {hashtag}
      </a>
    );
  }
}
