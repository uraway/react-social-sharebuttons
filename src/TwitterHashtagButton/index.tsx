import { FC, useEffect } from 'react';

declare let twttr: {
  widgets: {
    load: () => void;
  };
};

export type TwitterHashtagButtonProps = {
  hashtag: string;
  text?: string;
};

const TwitterHashtagButton: FC<TwitterHashtagButtonProps> = (props) => {
  useEffect(() => {
    let script: HTMLScriptElement;

    if (typeof twttr === 'undefined') {
      script = document.createElement('script');
      script.src = '//platform.twitter.com/widgets.js';
      script.async = true;
      script.id = 'twitter-wjs';
      document.getElementsByTagName('head')[0]?.appendChild(script);
    } else {
      twttr.widgets.load();
    }

    return () => script?.remove();
  }, []);
  const { hashtag, text } = props;
  return (
    <a
      href={`https://twitter.com/intent/tweet?button_hashtag=${hashtag}&text=${text || ''}`}
      className="twitter-hashtag-button"
    >
      Tweet {hashtag}
    </a>
  );
};

export default TwitterHashtagButton;
