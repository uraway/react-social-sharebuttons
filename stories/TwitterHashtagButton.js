import React from 'react';
import { text } from '@storybook/addon-knobs';
import { TwitterHashtagButton } from '../dist/index.esm.js';

export default {
  title: 'TwitterHashtagButton',
};

export const Basic = () => <TwitterHashtagButton hashtag={text('hashtag', 'sharebutton')} text={text('text', '')} />;
