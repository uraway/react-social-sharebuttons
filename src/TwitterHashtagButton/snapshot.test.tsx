import renderer from 'react-test-renderer';
import TwitterHashtagButton from '.';

it('renders correctly with required props', () => {
  const tree = renderer.create(<TwitterHashtagButton hashtag="hashtag" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with other props', () => {
  const tree = renderer.create(<TwitterHashtagButton hashtag="hashtag" text="text" />).toJSON();
  expect(tree).toMatchSnapshot();
});
