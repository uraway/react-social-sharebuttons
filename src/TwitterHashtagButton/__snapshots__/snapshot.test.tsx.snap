// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with other props 1`] = `
<a
  className="twitter-hashtag-button"
  href="https://twitter.com/intent/tweet?button_hashtag=hashtag&text=text"
>
  Tweet 
  hashtag
</a>
`;

exports[`renders correctly with required props 1`] = `
<a
  className="twitter-hashtag-button"
  href="https://twitter.com/intent/tweet?button_hashtag=hashtag&text="
>
  Tweet 
  hashtag
</a>
`;
